File: mobile/concord/lib/api/models/server_response.dart
import 'json_utils.dart';

class ServerResponse {
  const ServerResponse({
    required this.id,
    required this.name,
    required this.ownerId,
    required this.iconUrl,
    required this.created,
  });

  factory ServerResponse.fromJson(Map<String, dynamic> json) {
    return ServerResponse(
      id: json.field('Id') as String,
      name: json.field('Name') as String,
      ownerId: json.field('OwnerId') as String,
      iconUrl: json.field('IconUrl') as String?,
      created: parseDateTime(json.field('Created')),
    );
  }

  final String id;
  final String name;
  final String ownerId;
  final String? iconUrl;
  final DateTime created;
}

File: backend/Concord/Concord.Application/Models/ServerResponse.cs
using System.Text.Json.Serialization;

namespace Concord.Application.Models;

public class ServerResponse
{
    [JsonPropertyName("Id")]
    public Guid Id { get; set; }

    [JsonPropertyName("Name")]
    public string? Name { get; set; }

    [JsonPropertyName("OwnerId")]
    public Guid OwnerId { get; set; }

    [JsonPropertyName("IconUrl")]
    public string? IconUrl { get; set; }

    [JsonPropertyName("Created")]
    public DateTime Created { get; set; }

    [JsonPropertyName("JoinedNow")]
    public bool JoinedNow { get; set; }
}
